packages/next/next-devtools.webpack-config.js

```diff
@@ -0,0 +1,122 @@
+const path = require('path')
+const webpack = require('@rspack/core')
+const MODERN_BROWSERSLIST_TARGET = require('./src/shared/lib/modern-browserslist-target')
+const DevToolsIgnoreListPlugin = require('./webpack-plugins/devtools-ignore-list-plugin')
+
+function shouldIgnorePath(modulePath) {
+  // For consumers, everything will be considered 3rd party dependency if they use
+  // the bundles we produce here.
+  // In other words, this is all library code and should therefore be ignored.
+  return true
+}
+
+/**
+ * @param {Object} options
+ * @param {boolean} options.dev
+ * @param {Partial<webpack.Configuration>} options.rest
+ * @returns {webpack.Configuration}
+ */
+module.exports = ({ dev, ...rest }) => {
+  const experimental = false
+  const turbo = false
+
+  const bundledReactChannel = experimental ? '-experimental' : ''
+
+  const target = `browserslist:${MODERN_BROWSERSLIST_TARGET.join(', ')}`
+
+  return {
+    entry: path.join(
+      __dirname,
+      'src/client/components/react-dev-overlay/app/entrypoint.js'
+    ),
+    target,
+    mode: dev ? 'development' : 'production',
+    output: {
+      path: path.join(__dirname, 'dist/compiled/next-devtools'),
+      filename: `index.js`,
+      iife: false,
+      library: {
+        type: 'commonjs-static',
+      },
+    },
+    devtool: 'source-map',
+    optimization: {
+      moduleIds: 'named',
+      minimize: true,
+      concatenateModules: true,
+      minimizer: [
+        new webpack.SwcJsMinimizerRspackPlugin({
+          minimizerOptions: {
+            mangle: dev || process.env.NEXT_SERVER_NO_MANGLE ? false : true,
+          },
+        }),
+      ],
+    },
+    plugins: [
+      // TODO: React Compiler
+      new DevToolsIgnoreListPlugin({ shouldIgnorePath }),
+      new webpack.DefinePlugin({
+        // TODO: Hardcode it and ensure module resolution resolves to .node entrypoint in Node.js
+        // 'typeof window': JSON.stringify('object'),
+        'process.env.NEXT_MINIMAL': JSON.stringify('true'),
```

Comment:
`process.env.NEXT_MINIMAL` and `this.minimalMode` are not really needed for dev overlay, they shouldn't matter